File: Python/7py 1/ddy.py
"""
|--------------------------------------------------------------------------
| Simple Fun #144: Distinct Digit Year
|--------------------------------------------------------------------------
|
"""

def distinct_digit_year(year):
   x = 0
   while x != 4:
      year += 1
      arr = list(str(year))
      res = [arr.count(i) for i in arr]
      x = sum(res)
      if x == 4: return year


def distinct_digit_year(year):
   year += 1
   while len(set(str(year))) != 4: year += 1
   return year


from re import match
def distinct_digit_year(year):
   return year + 1 if match(r'(?:([0-9])(?!.*\1)){4}', str(year + 1)) else distinct_digit_year(year + 1)


print(distinct_digit_year(1987), 2013)
print(distinct_digit_year(2013), 2014)

File: PHP/7php 1/ddy.php
<?php
/*
|--------------------------------------------------------------------------
| Simple Fun #144: Distinct Digit Year
|--------------------------------------------------------------------------
|
*/

function distinctDigitYear($year) {
   $x = 0;
   while ($x != 4) {
      $year += 1;
      $arr = str_split(strval($year));
      $res = array_map(function($i) use ($arr) {
         return count(array_filter($arr, function($j) use ($i) {
            return $j == $i;
         }));
      }, $arr);
      $x = array_sum($res);
      if ($x == 4) return $year;
   }
}


function distinctDigitYear($year): int {
   return count(array_unique(str_split($year + 1))) == 4 ? $year + 1 : distinctDigitYear($year + 1);
}


function distinctDigitYear($year) {
   $year++;
   while(count(array_unique(str_split($year))) != 4) $year++;
   return $year;
}


function distinctDigitYear(int $year): int {
   return count(count_chars($year + 1, 1)) == 4 ? $year + 1 : distinctDigitYear($year + 1);
}


print_r(distinctDigitYear(1989) . ' ' . 2013 . '
');
print_r(distinctDigitYear(2013) . ' ' . 2014 . '
');